pkg/controller/status/status.go

```diff
@@ -670,8 +674,44 @@ func (m *statusManager) degradedReason() string {
 	return strings.Join(reasons, "; ")
 }
 
-func hasPendingCSR(ctx context.Context, cli client.Client, labelMap map[string]string) (bool, error) {
-	csrs := &v1beta1.CertificateSigningRequestList{}
+func hasPendingCSR(ctx context.Context, m *statusManager, labelMap map[string]string) (bool, error) {
+	if m.kubernetesVersion.Major > 1 || (m.kubernetesVersion.Major == 1 && m.kubernetesVersion.Minor >= 19) {
+		return hasPendingCSRUsingCertV1(ctx, m.client, labelMap)
+	}
+	return hasPendingCSRUsingCertV1beta1(ctx, m.client, labelMap)
```

Comment:
We should have a comment here explaining that we can remove this code once we stop supporting v1.18, so that future developers know when this can / cannot be removed.